v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -67,12 +67,9 @@ def from_ht(
         paths = dict(rdc_globals_struct.paths)
         versions = dict(rdc_globals_struct.versions)
         # enums are nested structs
-        enums = {k: dict(v) for k, v in rdc_globals_struct.enums.items()}
-
-        for global_dict in [paths, versions, enums]:
-            for dataset in list(global_dict.keys()):
-                if dataset not in datasets:
-                    global_dict.pop(dataset)
+        enums = {
+            k: dict(v) for k, v in rdc_globals_struct.enums.items() if k in datasets
```

Comment:
Do you think it's more accurate to filter to `datasets` or filter to `paths`?  I think they're gonna be the same in practice.  Just curious.

Reply:
I think I like filtering to `paths` better. If there is an erroneous enum not in `datasets`, at least we'll see one of those 'mismatch' logs for it which is more helpful for debugging